src/components/inscripcionTurnos/Paso1DDJJ.js

```diff
@@ -30,18 +31,30 @@ const AlertMessage = () => {
 };
 
 export default function Paso1DDJJ({ handleChange, agregarActividad }) {
+  const search = useLocation().search;
+  const ActividadID = new URLSearchParams(search).get('actividad');
+
   const actividades = useRecoilValue(actividadesUsuario());
   const cambioDeActividad = (nombre, actividad) => {
     agregarActividad(nombre, actividad);
   };
 
   const actividadesFiltradas = () => {
+    if (!ActividadID) {
+      return filter(
+        (actividad) =>
+          DateTime.fromISO(actividad.fechaHoraInicio) > DateTime.local(),
+        actividades
+      );
+    }
     return filter(
       (actividad) =>
-        DateTime.fromISO(actividad.fechaHoraInicio) > DateTime.local(),
+        DateTime.fromISO(actividad.fechaHoraInicio) > DateTime.local() &&
+        actividad.id == ActividadID,
       actividades
     );
   };
```

Comment:
Acá en vez de hacer 2 filter por separado, pudieron haber usado un ternario.

```suggestion
 const actividadesFiltradas = () => {
     const actividadesFuturas = DateTime.fromISO(actividad.fechaHoraInicio) > DateTime.local();
     
      return filter(
          (actividad) => 
              actividadID ? actividadesFuturas && actividad.id == ActividadID : actividadesFuturas, actividades
      );
}
```